mlflow/entities/span_event.py

```diff
@@ -22,3 +24,31 @@ class SpanEvent(_MLflowObject):
     # Use current time if not provided.
     timestamp: Optional[int] = field(default=int(time.time() * 1e6))
     attributes: Dict[str, Any] = field(default_factory=dict)
+
+
+class ExceptionEvent(SpanEvent):
+    """
+    An event that records an error during a span.
+    """
+
+    def _get_stacktrace(error: BaseException) -> str:
+        """Get the stacktrace of the parent error."""
+        msg = repr(error)
+        try:
+            if sys.version_info < (3, 10):
+                tb = traceback.format_exception(error.__class__, error, error.__traceback__)
+            else:
+                tb = traceback.format_exception(error)
+            return (msg + "\n\n".join(tb)).strip()
+        except Exception:
+            return msg
+
+    def __init__(self, exception: Exception, exception_stacktrace: Optional[str] = None):
+        self.name = "exception"
+        self.timestamp = int(time.time() * 1e6)
+        stack_trace = exception_stacktrace or self._get_stacktrace(exception)
+        self.attributes = {
+            "exception.message": str(exception),
+            "exception.type": exception.__class__.__name__,
+            "exception.stacktrace": stack_trace,
+        }
```

Comment:
```suggestion
    def __init__(self, exception: Exception, exception_stacktrace: Optional[str] = None):
        super().__init__(...)
```

we should use `super()` here.

Reply:
btw can we define a class method in `SpanEvent` instead of creating a new class?

Reply:
> btw can we define a class method in SpanEvent instead of creating a new class?

That sounds like a great idea to me, like `SpanEvent.from_exception()`.